### ISSUE TITLE
'describe-char' command leaves buffer in wrong Evil state

### ISSUE THREAD

**Issue by bertrandrustle** [syl20bnr/spacemacs#6768]
#### Description
After calling the `describe-char` command from a file buffer and then quitting the resulting _\*Help*_ window the cursor focus returns to the file buffer's window, but the cursor and window ID in the status line change colors to dark yellow, presumably indicating that Evil is no longer in the same state it was in before the command. From this new state, calling the `evil-normal-state` command seems to return the buffer to its original state and the cursor and window ID return to their normal cyan color.

Using the above procedure, the bug can't be triggered a second time in the same buffer and same Spacemacs session. It only happens once per buffer.

#### Reproduction guide
- Make sure  `dotspacemacs-editing-style` is set to `emacs` in your Spacemacs config
- Start Emacs
- Press **M-m f e d** to visit the `.spacemacs` file
- Move the cursor so that point is on any character in the file, then call the `describe-char` command with **M-m h d c**
- The cursor focus should move to the new _\*Help*_ window. From there, press **q** to dismiss the window, so that cursor focus returns to the `.spacemacs` window. The cursor and window ID should change to dark yellow.
- To return the file buffer to its original state, do **M-m : evil-normal-state** 

*Observed behaviour:*
In the `.spacemacs` buffer, the cursor and window ID in the status line change to a dark yellow color.

*Expected behaviour:*
The `.spacemacs` buffer should be in the same Evil state it was in before `describe-char` was called.

#### System Info
- OS: gnu/linux
- Emacs: 24.5.1
- Spacemacs: 0.105.21
- Spacemacs branch: master (rev. 0283f64)
- Graphic display: t
- Distribution: spacemacs
- Editing style: emacs
- Completion: helm
- Layers:
```elisp
(better-defaults emacs-lisp git markdown org
                 (shell :variables shell-default-height 30 shell-default-position 'bottom)
                 spell-checking version-control)
```

**Comment by TheBB** [comment]
Did you try reproducing this on develop? I'm unable to.

**Comment by bertrandrustle** [comment]
I just tried with a fresh installation from the develop branch, and I can't reproduce the bug.

**Comment by TheBB** [comment]
Thanks. Then I'm marking this as fixed in develop.

**Comment by d12frosted** [comment]
Fixed with released of Spacemacs v0.200. Let us know if you still have any problems with this issue.
